lib/pages/registration_page.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:flutter/material.dart';
+import 'package:tamini_app/components/custom_button.dart';
+import 'package:tamini_app/components/custom_text_field.dart';
+import 'package:tamini_app/components/otp_inputput_widget.dart';
+
+class RegistrationPage extends StatefulWidget {
+  const RegistrationPage({
+    super.key,
+  });
+
+  @override
+  State<RegistrationPage> createState() => _RegistrationPageState();
+}
+
+class _RegistrationPageState extends State<RegistrationPage> {
+  TextEditingController phoneNumberController = TextEditingController();
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  createNewUserFromMobile(BuildContext context, String phoneNumber) {
+    showDialog(
+      context: context,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          title: const Text('ادخل الرمز المرسل الى جوالك'),
+          content: OtpInputWidget(
+            onOtpEntered: (otp) {
+              // Handle the OTP verification logic here
+              if (otp == '1234') {
```

Comment:
Always write a valid logic and for bypassing use `kDebugMode`.
```suggestion
            onOtpEntered: (otp) {
              // Handle the OTP verification logic here
              if (kDebugMode) {
```